service-workers/service-worker/registration-updateviacache.https.html

```diff
@@ -0,0 +1,210 @@
+<!DOCTYPE html>
+<title>Service Worker: Registration-updateViaCache</title>
+<script src="/resources/testharness.js"></script>
+<script src="resources/testharness-helpers.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+  const UPDATE_VIA_CACHE_VALUES = [undefined, 'imports', 'all', 'none'];
+  const SCRIPT_URL = 'resources/update-max-aged-worker.py';
+  const SCOPE = 'resources/blank.html';
+
+  async function cleanup() {
+    const reg = await navigator.serviceWorker.getRegistration(SCOPE);
+    if (!reg) return;
+    if (reg.scope == new URL(SCOPE, location).href) {
+      return reg.unregister();
+    };
+  }
+
+  function wait(ms) {
+    return new Promise(r => setTimeout(r, ms));
+  }
+
+  function getScriptTimes(sw, testName) {
+    return new Promise(resolve => {
+      navigator.serviceWorker.addEventListener('message', function listener(event) {
+        if (event.data.test !== testName) return;
+        navigator.serviceWorker.removeEventListener('message', listener);
+        resolve({
+          mainTime: event.data.mainTime,
+          importTime: event.data.importTime
+        });
+      });
+
+      sw.postMessage('');
+    });
+  }
+
+  function registerViaApi(scriptUrl, opts) {
+    return navigator.serviceWorker.register(scriptUrl, opts);
+  }
+
+  function registerViaLinkElement(scriptUrl, opts) {
+    return new Promise((resolve, reject) => {
+      const link = document.createElement('link');
+
+      if (link.relList.supports('serviceworker') == false) throw Error("link rel=serviceworker not supported");
+
+      link.rel = 'serviceworker';
+      link.href = scriptUrl;
+      link.scope = opts.scope;
+
+      if (opts.updateViaCache) {
+        link.updateViaCache = opts.updateViaCache;
+      }
+
+      link.onload = async () => {
+        const fullScope = new URL(opts.scope, window.location).href;
+
+        const regs = await navigator.serviceWorker.getRegistrations();
+        const reg = regs.find(r => r.scope == fullScope && (r.installing || r.waiting || r.active));
+
+        if (reg) {
+          document.head.removeChild(link);
+          resolve(reg);
+        }
+        else {
+          reject(Error('Service worker not registered'));
+        }
+      };
+
+      document.head.appendChild(link);
+    });
+  }
+
+  async function registerViaLinkHeader(scriptUrl, opts) {
+    const link = document.createElement('link');
+
+    const fullScope = new URL(opts.scope, window.location).href;
+    scriptUrl = new URL(scriptUrl, location).href;
+
+    // Assuming that if this isn't supported, the header isn't
```

Comment:
Oh wow, now you point it out, that comment was really badly written. Fixed with your suggestion.